apps/aecore/lib/aecore/chain/chainstate.ex

```diff
@@ -20,6 +20,10 @@ defmodule Aecore.Chain.Chainstate do
   @protocol_version_field_size 64
   @protocol_version 17
 
+  @canonical_root_hash <<69, 176, 207, 194, 32, 206, 236, 91, 124, 28, 98, 196, 212, 25, 61, 56,
```

Comment:
could you explain this again?

Reply:
@thepiwo 
The way I understand it is the following.
When we create a PMT and no records are made, its root_hash is a empty binary. << >>
For some reason, perhaps related to the policy of development I do not know, in the elixir implementation of PMT, they do this check on root_hash and if it is empty, they have this additional logic on it and return PMT with this canonical_root_hash (ONLY if the RH is <<>>)
else the check is false and the origin PMT is returned.

Reply:
could you add a comment linking to the definition of this canonical root hash?

Reply:
@thepiwo 
This is the [function](https://github.com/aeternity/elixir-merkle-patricia-tree/blob/master/lib/trie.ex#L163-L174) which is called right after creating the PMT and deals with an empty root_hash at line 170

Reply:
<b>Correction</b> on https://github.com/aeternity/elixir-node/pull/484#discussion_r204439591
Not only if the RH is <<>>, <b> but also when the size of the resulting root_hash is greater than `max_rlp_len()` </b>

@thepiwo 
Which constant?

Reply:
`<<69, 176, 207, 194, 32, 206, 236, 91, 124, 28, 98, 196, 212, 25, 61, 56, 228, 235, 164, 142, 136, 21, 114, 156, 231, 95, 156, 10, 176, 228, 193, 192>>`

Reply:
@thepiwo 
This is the canonical root hash it is the same for all PMT if they are empty.

Reply:
can you add a comment in the file to explain that this is the root hash of an empty PMT?

Reply:
and lets wait for @meivantodorov opinion